core/block.js

```diff
@@ -1071,6 +1126,16 @@ Blockly.Block.prototype.setOutput = function(newBoolean, opt_check) {
   }
 };
 
+/**
+ * Set the maximum number of instances of this block allowed at a time.
+ * @param {number} maxInstances The maximum number of instances.
+ */
+Blockly.Block.prototype.setMaxInstances = function(maxInstances) {
```

Comment:
This should probably be on the workspace as a map of block type to max instances. If it's defined on each instance of a block what happens when different instances get different values?

Reply:
Makes sense to me. :D I moved it to be part of the Workspace.options object, because that seemed to make the most sense. My problem was there's no way to create a "reasonable default" for a map like this. What I ended up doing is leaving the maxInstances object as a public property on the options object, and then adding a "remainingCapacityOfType" function to the workspace. I'm not sure if there's a more readable/understandable configuration.